Recent versions of boost requiring C++14 ?

Installing and building on my laptop (Mac M3) with a fresh install, and ran into errors with `boost`.  (Possibly due to updates to `boost`?)

Lots of errors similar to this:
```
g++ -o source/main.o -c -std=c++11 -O3 -I/opt/local/include -I/usr/include/malloc -I/opt/homebrew/include source/main.cpp
In file included from source/main.cpp:8:
In file included from /opt/homebrew/include/boost/numeric/odeint.hpp:27:
In file included from /opt/homebrew/include/boost/numeric/odeint/stepper/euler.hpp:23:
/opt/homebrew/include/boost/numeric/odeint/stepper/base/explicit_stepper_base.hpp:165:42: error: 'auto' not allowed in lambda parameter
        m_resizer.adjust_size(in, [this](auto&& arg) { return this->resize_impl<StateIn>(std::forward<decltype(arg)>(arg)); });
                                         ^~~~
```

The error is due to `SConstruct` defining `-std=c++11`, and can be fixed by setting to `-std=c++14`.  [See this thread.](https://stackoverflow.com/questions/48653124/why-is-auto-not-allowed-in-lambda-parameter)

I know that the plan is to migrate away from `scons`, but this is really an issue with `boost`.  I think future installs might need to pin the C++ version (or the `boost` version, or both).

--- a/SConstruct
+++ b/SConstruct
@@ -18,7 +18,7 @@ AddOption('--linkflags', dest='linkflags', type='string', nargs=1, action='store
 
 
 subdirs = ['source']
-cxx_flags = ['-std=c++11',]
+cxx_flags = ['-std=c++14',]
 try:
     CXX = os.environ['CXX']
 except KeyError:

